src/components/topNav/TopNav.tsx

```diff
@@ -1,46 +1,55 @@
 import * as React from "react";
-import * as styles from "./TopNav.module.css";
 import { Link } from "@gemeente-denhaag/components-react";
+import * as styles from "./TopNav.module.css";
 import { navigate } from "gatsby";
+import clsx from "clsx";
 
 export interface ITopNavItem {
   label: string;
   href: string;
   icon?: JSX.Element;
+  linkState?: any;
 }
 
-interface TopNavItemsProps {
+interface TopNavProps {
   items: ITopNavItem[];
+  layoutClassName?: string;
 }
 
-export const PrimaryTopNav: React.FC<TopNavItemsProps> = ({ items }) => {
+export const PrimaryTopNav: React.FC<TopNavProps> = ({ items, layoutClassName }) => {
   return (
-    <nav className={styles.primary}>
-      <ul className={styles.ul}>
-        {items.map(({ label, href, icon }, idx) => (
-          <li className={styles.li} key={idx} onClick={() => navigate(href)}>
-            <Link icon={icon} iconAlign="start">
-              {label}
-            </Link>
-          </li>
-        ))}
-      </ul>
-    </nav>
+    <div className={clsx([styles.primary, layoutClassName && layoutClassName])}>
+      <nav className={styles.primary}>
+        <ul className={styles.ul}>
+          {items.map(({ label, href, icon, linkState }, idx) => (
+            <li className={styles.li} key={idx} onClick={() => navigate(href, { state: linkState })}>
```

Comment:
Instead of making our own `onClick` function here, we'd just pass the `handleClick` that was set by its context.